### PR TITLE
[AI-FSSDK] [FSSDK-12337] Add Feature Rollout support

### PR DESCRIPTION
**Pull request by FarhanAnjum-opti**
## Summary

Adds Feature Rollout support to the JavaScript SDK. Feature Rollouts are a new experiment rule type that combines Targeted Delivery simplicity with A/B test measurement capabilities. During project config parsing, the "everyone else" variation from the flag's rollout is injected into any experiment with type "feature_rollout", enabling correct evaluation without changes to decision logic.

## Changes

- Added optional `type` string field to the Experiment interface
- Added config parsing logic to inject the "everyone else" rollout variation into feature_rollout experiments
- Added traffic allocation entry (endOfRange=10000) for the injected variation
- Added `getEveryoneElseVariation` helper function to extract the last rollout rule's first variation
- Updated variation lookup maps (variationKeyMap, variationIdMap, variationVariableUsageMap) with injected variation
- Added 6 unit tests covering feature rollout injection, edge cases, and backward compatibility

## Jira Ticket

[FSSDK-12337](https://optimizely-ext.atlassian.net/browse/FSSDK-12337)

[FSSDK-12337]: https://optimizely-ext.atlassian.net/browse/FSSDK-12337?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/project_config/project_config.ts [hunk 1]
```diff
@@ -260,6 +260,18 @@ export const createProjectConfig = function(datafileObj?: JSON, datafileStr: str
   projectConfig.experimentKeyMap = keyBy(projectConfig.experiments, 'key');
   projectConfig.experimentIdMap = keyBy(projectConfig.experiments, 'id');
 
+  const validExperimentTypes = new Set(Object.values(EXPERIMENT_TYPES));
+  (projectConfig.experiments || []).forEach(experiment => {
+    if (experiment.type != null && !validExperimentTypes.has(experiment.type)) {
+      throw new OptimizelyError(
```
Comment (raju-opti) [inline lib/project_config/project_config.ts]: this should not throw if the experiment type is unrecognized, otherwise this version of the SDK will throw errors if a new rule type is added in future datafiles

File: lib/project_config/project_config.ts [hunk 2]
```diff
@@ -301,6 +313,30 @@ export const createProjectConfig = function(datafileObj?: JSON, datafileStr: str
...
+      const experiment = projectConfig.experimentIdMap[experimentId];
+      if (experiment && experiment.type === EXPERIMENT_TYPES.FR) {
+        experiment.variations.push(everyoneElseVariation);
+        experiment.trafficAllocation.push({
+          entityId: everyoneElseVariation.id,
+          endOfRange: 10000,
+        });
+        // Update variation lookup maps
+        experiment.variationKeyMap[everyoneElseVariation.key] = everyoneElseVariation;
+        projectConfig.variationIdMap[everyoneElseVariation.id] = everyoneElseVariation;
```
Comment (raju-opti) [inline lib/project_config/project_config.ts]: projectConfig.variationIdMap and projectConfig.variationVariableUsageMap should already contain the correct values. We can remove these two lines

**Review by raju-opti — CHANGES_REQUESTED** [review]

**Review by raju-opti — APPROVED** [review]
